packages/dashboard/src/CdcDashboard.tsx

```diff
@@ -219,14 +310,15 @@ export default function CdcDashboard({ configUrl = '', config: configObj = undef
   }
 
   const loadConfig = async () => {
-    let response = configObj || (await (await fetch(configUrl)).json())
+    setLoading(true)
+    let response: Config = configObj || (await (await fetch(configUrl)).json())
     let newConfig = { ...defaults, ...response }
-    let datasets = {}
```

Comment:
@adamdoe here datasets is an Object, setData is later called and the object is passed.